pkg/kds/v2/reconcile/reconciler.go

```diff
@@ -15,24 +15,20 @@ import (
 	core_model "github.com/kumahq/kuma/pkg/core/resources/model"
 	cache_v2 "github.com/kumahq/kuma/pkg/kds/v2/cache"
 	util_kds_v2 "github.com/kumahq/kuma/pkg/kds/v2/util"
+	"github.com/kumahq/kuma/pkg/multitenant"
 	"github.com/kumahq/kuma/pkg/util/xds"
 )
 
 var log = core.Log.WithName("kds-delta").WithName("reconcile")
 
-func NewReconciler(
-	hasher envoy_cache.NodeHash,
-	cache envoy_cache.SnapshotCache,
-	generator SnapshotGenerator,
-	mode config_core.CpMode,
-	statsCallbacks xds.StatsCallbacks,
-) Reconciler {
+func NewReconciler(hasher envoy_cache.NodeHash, cache envoy_cache.SnapshotCache, generator SnapshotGenerator, mode config_core.CpMode, statsCallbacks xds.StatsCallbacks, hashingFn multitenant.HashingFn) Reconciler {
```

Comment:
Might be worth opening an issue for `envoy_cache.NodeHash` to take in the context no? That way we wouldn't have to pass 2 hashing Fn to the Reconciler